api/app/signals/apps/api/tests/test_private_signal_endpoint.py

```diff
@@ -1454,6 +1458,153 @@ def test_missing_email_or_phone_as_empty_string(self):
         self.assertEqual(response_json['reporter']['email'], '')
         self.assertEqual(response_json['reporter']['phone'], '')
 
+    def test_no_permissions_to_view_contact_details(self):
+        """
+        SIG-4160 - [BE] Rechten op contactgegevens
+
+        When a user has no "sia_can_view_contact_details" and there is an email and/or phone they should be rendered as
+        "*****". If there is no email and/or phone they should still render as "".
+        """
+        detail_endpoint = self.detail_endpoint.format(pk=self.signal_with_image.id)
+
+        self.signal_with_image.reporter.email = 'm.elder@example.com'
+        self.signal_with_image.reporter.phone = '0123456789'
+        self.signal_with_image.reporter.save()
+
+        response = self.client.get(detail_endpoint)
+        response_json = response.json()
+        self.assertEqual(response_json['reporter']['email'], '*****')
+        self.assertEqual(response_json['reporter']['phone'], '*****')
+
+        self.signal_with_image.reporter.email = 'm.elder@example.com'
+        self.signal_with_image.reporter.phone = None
+        self.signal_with_image.reporter.save()
+
+        response = self.client.get(detail_endpoint)
+        response_json = response.json()
+        self.assertEqual(response_json['reporter']['email'], '*****')
+        self.assertEqual(response_json['reporter']['phone'], '')
+
+        self.signal_with_image.reporter.email = None
+        self.signal_with_image.reporter.phone = '0123456789'
+        self.signal_with_image.reporter.save()
+
+        response = self.client.get(detail_endpoint)
+        response_json = response.json()
+        self.assertEqual(response_json['reporter']['email'], '')
+        self.assertEqual(response_json['reporter']['phone'], '*****')
+
+        self.signal_with_image.reporter.email = None
+        self.signal_with_image.reporter.phone = None
+        self.signal_with_image.reporter.save()
+
+        response = self.client.get(detail_endpoint)
+        response_json = response.json()
+        self.assertEqual(response_json['reporter']['email'], '')
+        self.assertEqual(response_json['reporter']['phone'], '')
```

Comment:
Add test for missing email / phone number represented in DB as `""` (these exist in old nuisance complaints).

Reply:
Added test